app/models/event_topic.py

```diff
@@ -20,20 +22,31 @@ class EventTopic(SoftDeletionModel):
     __tablename__ = 'event_topics'
 
     id = db.Column(db.Integer, primary_key=True, autoincrement=True)
-    name = db.Column(db.String, nullable=False)
+    name = db.Column(db.String, nullable=True)
+    system_image_url = db.Column(db.String)
     slug = db.Column(db.String, unique=True, nullable=False)
     events = db.relationship('Event', backref='event_topics')
     event_sub_topics = db.relationship('EventSubTopic', backref='event-topic')
 
     def __init__(self,
                  name=None,
+                 system_image_url=None,
                  slug=None,
                  deleted_at=None):
 
         self.name = name
+        self.system_image_url = self.set_default_sys_image() \
+            if system_image_url is None else system_image_url
         self.slug = get_new_slug(name=self.name)
         self.deleted_at = deleted_at
 
+    @classmethod
+    def set_default_sys_image(cls):
+        with app.test_request_context():
+            url = urllib.parse.urlparse(request.url)
+            image_url = url.scheme + '://' + url.netloc + '/static/header.png'
+            return image_url
```

Comment:
This is test_request context

Pass real request here

Reply:
app.request_context() requires argument `environ`
```
with app.request_context():
            url = urllib.parse.urlparse(request.url)
            image_url = url.scheme + '://' + url.netloc + '/static/header.png'
            return image_url
```
Even it is test_request_context() it works fine. Also, RequestContexttask using it  https://github.com/fossasia/open-event-server/blob/development/app/api/helpers/request_context_task.py#LC41

Reply:
Celery tasks are separate things because they are detached from actual requests, here you are sending response from a request, why can't you simply take that request and pass it here?

Reply:
without using `with app.test_request_context():` it's not working.

Reply:
without `test_request_context` error comes
```
    raise RuntimeError(_request_ctx_err_msg)
RuntimeError: Working outside of request context.

This typically means that you attempted to use functionality that needed
an active HTTP request.  Consult the documentation on testing for
information about how to avoid this problem.
```